private/model/api/eventstream_tmpl.go

```diff
@@ -214,6 +214,13 @@ func (es *{{ $esapi.Name }}) waitStreamPartClose() {
 			es.inputWriter = inputWriter
 	}
 
+	func (es *{{ $esapi.Name }}) closeInputWriter(r *request.Request) {
+		err := es.inputWriter.Close()
+		if err != nil {
+			r.Error = awserr.New(eventstreamapi.InputWriterCloseErrorCode, err.Error(), r.Error)
+		}
+	}	
```

Comment:
This does create a minor bifurcation in how closing the InputWriter is done in success vs failure cases. Is there anyway to merge this with the success exit path? This is something that seems like it would be better as a function closure instead of method on the `$esapi.Name` type. 

Can the `es.Close` not be used instead of this method?

Reply:
`es.Close` does more stuff than we require, and is good to use after streaming starts. `Close` calls to `safeClose` - which tries to close es.Writer and es.OutputReader. These two are nil - for the first request until we setup and start streaming - which happens in unmarshal handler step. In case of error response, the writer/reader are never initialized.
[https://github.com/aws/aws-sdk-go/blob/main/service/transcribestreamingservice/api.go#L722](https://github.com/aws/aws-sdk-go/blob/main/service/transcribestreamingservice/api.go#L722)

I thought of changing Close() to use it, but it doesn’t gain us much. Having this separate makes it simpler and explicit imo, and we do not have to worry about adding nil checks in Close (which the original design seems to have tried not to add).

Reply:
updated naming from `closeInputWriter` to `closeInputPipe` as suggested.